OpenSim/Simulation/Model/Geometry.h

```diff
@@ -180,8 +150,34 @@ class OSIMSIMULATION_API Geometry : public Component {
     /// Methods to support frame as a connection, implement Component interface
     void constructConnectors() override;
 
+protected:
+    /// Map this Geometry into a list of primitives aka SimTK::DecorativeGeometry 
+    /// and return the result in the passed in Array.
+    virtual void createDecorativeGeometry(
+        SimTK::Array_<SimTK::DecorativeGeometry>&) const = 0;
+
 private:
-    
+    /// Compute Transform of this geometry relative to its base frame, utilizing 
+    /// passed in state. Both transform and body_id are set in the passed-in 
+    /// decorations as a side effect.
+    void setDecorativeGeometryTransform(
```

Comment:
I still don't really know what this method does. The comment is confusing since a geometry is fixed to a frame and that frame's base frame can be accessed directly, so what is being computed here? As a private method you don't need the doxygen style comments, since i) we don't doc private members in general and ii) even if we did allow some, you probably would not want these methods to be known, correct?

Is this method setting the transforms of the drawing primitives used by the visualizer? As a private method, a user doesn't need to know about it, but it may be helpful for whoever has to maintain it to have some clear tips.